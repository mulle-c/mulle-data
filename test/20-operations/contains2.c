#include <mulle-data/mulle-data.h>

#include <stdio.h>

static void  mulle_range_print( struct mulle_range a)
{
   printf( "{ %td, %td }", a.location, a.length);
}


int   main( int argc, char *argv[])
{
   int                  flag;
   struct mulle_range   a;
   struct mulle_range   b;

   a    = mulle_range_make( 0, 10);
   b    = mulle_range_make( 5, 0);
   flag = mulle_range_contains( a, b);
   mulle_range_print( a);
   printf( " contains ");
   mulle_range_print( b);
   printf( " = %s\n", flag ? "YES" : "NO");

   return( 0);
}


/*
 * extension : mulle-sde/c-test-library-demo
 * directory : demo/all
 * template  : .../hello.c
 * Suppress this comment with `export MULLE_SDE_GENERATE_FILE_COMMENTS=NO`
 */
