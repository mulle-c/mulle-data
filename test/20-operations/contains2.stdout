{ 0, 10 } contains { 5, 0 } = YES
